### DISCUSSION TITLE
How to use custome environment variables (without hard coding the values in the config files)

### DISCUSSION THREAD

**Discussion by Fettah** [renovatebot/renovate#18568] [Request Help]
I am using github actions abd I am storing some value as secret which will  be available as environment variables. I want to access these variables form my renovate config.js files
    `process.ENV.VARIABLE_NAME` does not seem to work

There seems to be a PR that introduced this features but it is not document how it shall be used: https://github.com/renovatebot/renovate/pull/8321/files#

Here is my renovate-config.js file:
```

    module.exports = {
      platform: 'github',
      logLevel: 'debug',
      labels: ['renovate', 'dependencies', 'automated'],
      onboarding: true,
      onboardingConfig: {
        extends: ['config:base', 'disableDependencyDashboard']
      },
      cacheDir: "/tmp/renovate",
      renovateFork: true,
      gitAuthor: "renovate <renovate@hhpv.de>",
      username: "Renovate",
      onboarding: false,
      printConfig: true,
      requireConfig: false,
      logLevel: "DEBUG",
      baseBranches: ["ecr-renovate"],
      customEnvVariables: {
        // what should i put here
      },
      hostRules: [
        {
          hostType: 'docker',
          matchHost: '123456789.dkr.ecr.eu-central-1.amazonaws.com',
          //username: process.env.AWS_ACCESS_KEY,
          //password: process.env.AWS_SECRET_KEY
        },
      ],
    };
```

**Comment by viceice** [comment]
What is `github actions abd` ?

Are you using our official github action? You can't use arbitrary env values there.

`customEnvVariables` is a simple dictioniary, for passing additional variables to tools called by renovate. those arn't used by renovate itself.

**Reply by Fettah** [reply]
yes i am using the official github action and I need to pass some secrets that are stored in github

**Reply by viceice** [reply]
you can't pass arbitrary env values, but you can use `detectHostRulesFromEnv`[^1]

All `RENOVATE_` variables are automatically passed

https://github.com/renovatebot/github-action/blob/9edb599d851b00ab5c5e6abe9443fd6ced5a5724/src/input.ts#L11

[^1]: https://docs.renovatebot.com/self-hosted-configuration/#detecthostrulesfromenv
